sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,41 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainModuleInitializer := {
+        if (scalaJSUseMainModuleInitializer.value) {
+          mainClass.value.fold {
+            throw new MessageOnlyException(
+                "No main class class detected, but " +
+                "scalaJSUseMainModuleInitializer was set to true. If you do " +
+                "not want to derive a main module initializer from " +
+                "discovered main classes, add " +
+                "`scalaJSMainModuleInitializer := None`")
+          } { mainClass =>
+            Some(ModuleInitializer.mainMethod(mainClass, "main"))
+          }
+        } else {
+          None
+        }
+      },
+
+      scalaJSModuleInitializers ++= scalaJSMainModuleInitializer.value.toList,
```

Comment:
This instead should check whether we want to *use* the main module initializer and conditionally add it.